shared/chat/shared.js

```diff
@@ -0,0 +1,13 @@
+// @flow
+import * as Constants from '../constants/chat'
+import createCachedSelector from 're-reselect'
+
+const lookupMessageProps = createCachedSelector(
```

Comment:
This selector is now used in 3 places, making better use of caching via re-reselect.